### PR TITLE
Fixed bug with formatting code in appcode mac os

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/mprobst/ClangFormatIJ/issues/1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 .idea/workspace.xml
 out
-ClangFormatIJ.jar
```
Comment (mprobst) [inline .gitignore:3]: Why do you remove this?

File: .idea/misc.xml [hunk 2]
```diff
@@ -51,7 +51,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="IntelliJ IDEA Community Edition IC-145.972.3" project-jdk-type="IDEA JDK">
```
Comment (mprobst) [inline .idea/misc.xml:54]: Is this required by anything?

File: README.md [hunk 3]
```diff
@@ -5,7 +5,13 @@ IDEA, WebStorm, or similar.
...
 
-You will also need `clang-format` itself. Install it from npm (`npm install
+Install plugin from disk >>
+https://www.jetbrains.com/help/idea/2016.3/installing-plugin-from-disk.html
+
+You will also *need* `clang-format` itself. Install it from npm (`npm install
 clang-format`), or using homebrew (`brew install clang-format`).
+
+## Screenshots
+![Clang-screenshot](https://1.downloader.disk.yandex.ru/disk/550427a1ed241178d4db76075e852182f9082c5b27e7e03662ab5d4c8533e7c4/58c200e6/fKqInKw3d7bLFOeFnMGnhNRkwkoEzPUnyeOWDcOml95JJux8diRxVJEXTokE8N2irBe6mEDGrfoX6hPRTbyMVC1bnVzdXZfCiH_li6hDcwWr8npumZHI4midPdWhecNq?uid=0&filename=screenshot%202.png&disposition=inline&hash=&limit=0&content_type=image%2Fpng&fsize=115477&hid=637d6c440b625c0f3064ac18690801c4&media_type=image&tknv=v2&etag=487e1ecb31fc7fd67a8a1b757f5ee31d?raw=true "Screenshot")
```
Comment (mprobst) [inline README.md:17]: this URL gives me a 410?

File: README.md [hunk 4]
```diff
@@ -5,7 +5,13 @@ IDEA, WebStorm, or similar.
 
 ## Installation
 
-It's easiest to install ClangFormatIJ from the plugin marketplace.
+Download plugin from https://github.com/CheslavBochko/ClangFormatIJ/blob/master/Plugin/ClangFormatIJ.jar
```
Comment (mprobst) [inline README.md:8]: let's keep the existing plugin location, shouldn't we?

File: resources/META-INF/plugin.xml [hunk 5]
```diff
@@ -39,15 +40,17 @@
...
-    </action>
+    <group id="ClangFormat.Group" text="ClangFormat" popup="true">
+      <action id="clangFormat" class="io.probst.idea.clangformat.ClangFormatAction" text="_Format this file"
+              description="Format this file">
+        <keyboard-shortcut keymap="Mac OS X" first-keystroke="control alt K"/>
+        <keyboard-shortcut keymap="Mac OS X" first-keystroke="meta alt K"/>
+        <keyboard-shortcut keymap="Mac OS X 10.5+" first-keystroke="meta alt K"/>
+
+      </action>
+      <add-to-group group-id="CodeMenu"/>
```
Comment (mprobst) [inline resources/META-INF/plugin.xml:51]: What's the difference between CodeMenu and EditorMenu? Should we add this to both?

File: resources/META-INF/plugin.xml [hunk 6]
```diff
@@ -39,15 +40,17 @@
...
-    <action id="ClangFormat.ClangFormat" class="io.probst.idea.clangformat.ClangFormatAction"
-            text="clang-format"
-            description="Runs clang-format on this buffer">
-      <add-to-group group-id="EditorPopupMenu" anchor="last"/>
-
-      <keyboard-shortcut keymap="Mac OS X" first-keystroke="control alt K"/>
-      <keyboard-shortcut keymap="Mac OS X" first-keystroke="meta alt K"/>
-      <keyboard-shortcut keymap="Mac OS X 10.5+" first-keystroke="meta alt K"/>
-    </action>
+    <group id="ClangFormat.Group" text="ClangFormat" popup="true">
```
Comment (mprobst) [inline resources/META-INF/plugin.xml:43]: Do we need a group here? It seems that we only have one action in it, wouldn't this cause a somewhat odd `ClangFormat -> Format this file` menu structure? Just one level would be easier, wouldn't it?

File: src/io/probst/idea/clangformat/ClangFormatAction.java [hunk 7]
```diff
@@ -65,19 +69,12 @@ public void actionPerformed(AnActionEvent actionEvent) {
     // clang-format.
     int docLength = document.getTextLength() - 1;
     int cursor = Math.min(caret.getOffset(), docLength);
-    int selectionStart = Math.min(editor.getSelectionModel().getSelectionStart(), docLength);
```
Comment (mprobst) [inline src/io/probst/idea/clangformat/ClangFormatAction.java:68]: with this change, you can no longer change an individual selection range, can you?

File: src/io/probst/idea/clangformat/ClangFormatAction.java [hunk 8]
```diff
@@ -165,10 +158,7 @@ public String call() throws Exception {
   private ProcessBuilder getCommand(String filePath, int cursor, int selectionStart,
       int selectionLength) throws ExecutionException, InterruptedException {
     Settings settings = Settings.get();
-
-    ProcessBuilder command = new ProcessBuilder().command(settings.clangFormatBinary, "-style=file",
-        "-output-replacements-xml", "-assume-filename=" + filePath, "-cursor=" + cursor,
-        "-offset=" + selectionStart, "-length=" + selectionLength);
+    ProcessBuilder command = new ProcessBuilder().command(settings.clangFormatBinary,filePath, "-style=file", "-cursor=" + cursor);
```
Comment (mprobst) [inline src/io/probst/idea/clangformat/ClangFormatAction.java:161]: This is a pretty substantial change. It means clang-format only formats the file on disk - all the offset handling above or ranges will no longer work, and you also can no longer format modified buffers without writing them first. I don't think this is a good change, TBH.

Didn't you say you found an issue where it just calculated an offset of a character incorrect wrt the encoding?

Comment (ghost) [reply src/io/probst/idea/clangformat/ClangFormatAction.java:161]: All right. I didn't find a quick solution. That's why I wrote it as a temporary solution.

**Pull request closed without merging**